File: 1493_longest_subarray_of_1_after_deleting_one_element/longestSubarray.go
package main

func longestSubarray(nums []int) int {
	l, r := 0, 0
	c := 0

	for r < len(nums) {
		c += 1 - nums[r]
		if c > 1 {
			c -= 1 - nums[l]
			l++
		}

		r++
	}

	return r - l - 1
}

File: 1493_longest_subarray_of_1_after_deleting_one_element/longestSubarray.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        l, r = 0, 0
        c = 0

        while r < len(nums):
            c += 1 - nums[r]

            if c > 1:
                c -= 1 - nums[l]
                l += 1

            r += 1

        return r - l - 1
